tests/dialects/test_tsql.py

```diff
@@ -2090,3 +2090,27 @@ def test_next_value_for(self):
                 "oracle": "SELECT NEXT VALUE FOR db.schema.sequence_name",
             },
         )
+
+    # string literals in the DATETRUNC are casted as DATETIME2
```

Comment:
I don't think that'd be a valuable comment @VaggelisD, the builder code is easy to follow.

Reply:
It might look like an incorrect assumption considering this is `DATE_TRUNC` and the string is being casted to `DATETIME2`, but yeah no strong preference on it

Reply:
if you wanna leave a comment in date trunc's builder, then put a reference to the doc page [where it says](https://learn.microsoft.com/en-us/sql/t-sql/functions/datetrunc-transact-sql?view=sql-server-ver16#date) that if `date` is a string, it needs to be resolvable to a datetime2:

> DATETRUNC also accepts a string literal (of any string type) that can resolve to a datetime2(7).